mesheryctl/internal/cli/root/registry/publish.go

```diff
@@ -0,0 +1,129 @@
+// # Copyright Meshery Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package registry
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	// "github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	// "github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	// "github.com/pkg/errors"
+
+	// log "github.com/sirupsen/logrus"
+
+
+	"github.com/spf13/cobra"
+	// "github.com/spf13/viper"
+)
+
+var (
+	system string
+	googleSheetCredential string
+	sheetID string
+	modelsOutputPath string
+	imgsOutputPath string
+)
+
+// publishCmd represents the publish command to publish Meshery Models to Websites, Remote Provider, Meshery
+var publishCmd = &cobra.Command{
+	Use:   "publish [system] [google-sheet-credential] [sheet-id] [models-output-path] [imgs-output-path]",
+	Short: "Publish Meshery Models to Websites, Remote Provider, Meshery",
+	Long:  `Publishes metadata about Meshery Models to Websites, Remote Provider and Meshery by reading from a Google Spreadsheet.`,
+	Example: `
+// Publish To System
+mesheryctl publish [system] [google-sheet-credential] [sheet-id] [models-output-path] [imgs-output-path]
+
+// Publish To Meshery
+mesheryctl registry publish meshery GoogleCredential GoogleSheetID <repo>/server/meshmodel
```

Comment:
As we have moved this command to `exp`, it should now be `mesheryctl exp publish`.